### PR TITLE
ENT-8227 | Throw exception if lowest numbered server does not exist.

### PR DESCRIPTION
**Pull request by webbj74**
I believe in the occasions when we ask for the lowest numbered server, we expect it to be there. See JIRA ticket for example output in a downstream application.

### CODE REVIEW

File: src/Hosting/Server/ServerList.php [hunk 1]
```diff
@@ -105,6 +106,12 @@ public function getLowestNumberedServer()
         $lowestServer = null;
         $lowestId = 999999;
 
+        if (!$this->count()) {
+            throw new RuntimeException(
```
Comment (pbull) [inline src/Hosting/Server/ServerList.php]: Just spitballing... would it be useful to define a custom error class (e.g. `EmptyServerListException`) to make it easier to catch this specific case in code that calls this method?

File: tests/Hosting/Server/ServerListTest.php [hunk 2]
```diff
@@ -20,6 +20,7 @@
 use Acquia\Platform\Cloud\Hosting\Server\WebServerListInterface;
 use InvalidArgumentException;
 use PHPUnit\Framework\TestCase;
+use Acquia\Platform\Cloud\Hosting\Server\EmptyServerListException;
```
Comment (pbull) [inline tests/Hosting/Server/ServerListTest.php]: np: ordering of use statements

**Review by pbull — COMMENTED** [review]
Overall looks good, one minor nitpick.

File: tests/Hosting/Server/ServerListTest.php (hunk 2 above)
Comment (webbj74-acquia) [reply tests/Hosting/Server/ServerListTest.php]: thanks; fixed.

**Review by pbull — APPROVED** [review]

**Review by JinaLeeK — APPROVED** [review]

**Pull request merged**